networkx/algorithms/bipartite/matching.py

```diff
@@ -582,3 +585,310 @@ def minimum_weight_full_matching(G, top_nodes=None, weight="weight"):
     # add the ones from right to left as well.
     d.update({v: u for u, v in d.items()})
     return d
+
+
+def _neighbours_of_set(G, node_set):
+    r"""Returns a set of the neighbours of a given set of nodes `node_set` in the bipartite graph `G`.
+    Denoted by :math:`N_Gnode_set`.
+
+    `G` is an undirected bipartite NetworkX graph.
+
+    `node_set` is a set of nodes in `G`.
+
+
+    >>> G = nx.complete_bipartite_graph(3,3)
+    >>> _neighbours_of_set(G, {})
+    set()
+    >>> _neighbours_of_set(G, {1, 2})
+    {3, 4, 5}
+
+    >>> G = nx.Graph([(0, 4), (1, 5), (2, 6)])
+    >>> _neighbours_of_set(G, {0, 1})
+    {4, 5}
+
+    >>> G=nx.Graph([(0,3),(3,0),(0,4),(4,0),(1,4),(4,1),(2,4),(4,2)])
+    >>> _neighbours_of_set(G, {0, 1})
+    {3, 4}
+
+    >>> _neighbours_of_set(G, {4})
+    {0, 1, 2}
+    """
+    ret_set = {}
+    for node in node_set:
+        ret_set.update(G[node])
+
+    return set(ret_set)
```

Comment:
This code will just work fine, but you can speed this bit up with something like :)
```suggestion
    # add import at the top of the file
    from itertools import chain
    neighbors_iter = chain.from_iterable(G.neighbors(n) for n in node_set)
    return set(neighbors_iter)
```

Reply:
I see! Nice use of itertools, I'll add it as a separate commit.
BTW, maybe we can add this function not inside bipartite/matching.py but instead make it available to the whole package, I'm pretty sure someone else will find usage for this in the future.
To take it a step further maybe we can modify Graph.neighbors to take in an iterable effectively extending the functionality of it. I'm guessing if you think this is a good idea maybe I can create a separate PR for this, as this isn't a function only used for the calculation of envy-free bipartite matchings, and in general the neighbors set is something that's very common in graph theory and so if this can be of help to someone in the future, why not!

Reply:
I went ahead and added this in a commit.
I noticed when looking at the imports that there is a 'import itertools' line so I didn't import chain specifically as you noted, instead, I called itertools.chain -  I thought it is a good practice to avoid as much as possible editing these common grounds in the code to avoid merging issues, correct me if I'm wrong.